backend/src/main/java/com/huellapositiva/domain/service/ProposalService.java

```diff
@@ -107,4 +110,23 @@ public ProposalRevisionEmail requestChanges(
     private boolean hasFeedback(ProposalRevisionDto proposalRevisionDto) {
         return proposalRevisionDto.getFeedback() != null;
     }
+
+    /**
+     * This method find the proposal in the database and checks if the status is PUBLISHED for
+     * close the enrollment. Otherwise, a ProposalNotClosableException with response status 409 will be throw.
+     * @param idProposal : The id of the proposal to be checked and updated.
+     */
+    public void changeStatusToEnrollmentClosed(String idProposal) {
+        JpaProposal proposal = jpaProposalRepository.findByNaturalId(idProposal).orElseThrow(EntityNotFoundException::new);
+        String status = proposal.getStatus().getName().toUpperCase();
+
+        if (!PUBLISHED.toString().equals(status)) {
+            throw new ProposalNotClosableException();
```

Comment:
The proposal does not get closed. What it does get closed is the enrollment of the proposal.

ProposalEnrollmentNotCloseableException/EnrollmentNotCloseableException